test/open-uri/test_open-uri.rb

```diff
@@ -115,6 +116,13 @@ def test_open_too_many_arg
     assert_raise(ArgumentError) { open("http://192.0.2.1/tma", "r", 0666, :extra) {} }
   end
 
+  def test_tempfile_rest_parameter
+    temp_file = Tempfile.new
+    assert_nothing_raised { open(temp_file, 'a') }
```

Comment:
I'm not sure how this workaround should fix this bug? IMHO requiring `open-uri` should not break how `open` works but only add new functionalities.

Just some background information: We added a new dependency to our project which broke some completely different code which has nothing to do with the new dependency. Our tests noticed this. After some debugging I found out that the dependency had a dependency which ...  required `open-uri`. Things like this should not happen.